### PR TITLE
Report modlog

### PR DESCRIPTION
**Pull request by weeryan17**
- [x] Completed PR - Has all features or fixes that it was meant to and isn't half done
- [x] Follows the style guide - Style guide found [here](https://github.com/FlareBot/FlareBot/blob/master/CONTRIBUTING.md)
- [x] All changes have been tested

## Added
Adding modlog logging for reports

## Changed
N/A

## Deprecated
N/A

## Removed
N/A

## Fixed
N/A

## Security
N/A

### CODE REVIEW

File: src/main/java/stream/flarebot/flarebot/commands/moderation/ReportsCommand.java [hunk 1]
```diff
@@ -114,8 +115,17 @@ public void onCommand(User sender, GuildWrapper guild, TextChannel channel, Mess
...
+                            ReportStatus old = guild.getReportManager().getReport(id).getStatus();
                             guild.getReportManager().getReport(id).setStatus(status);
                             MessageUtils.sendSuccessMessage(String.format("Changed status of Report with ID: **%d** to **%s**", id, status.getMessage()), channel, sender);
+                            EmbedBuilder eb = new EmbedBuilder();
+                            eb.setTitle("Report edited");
+                            eb.setColor(Color.WHITE);
+                            eb.addField("Report ID", String.valueOf(id), true);
+                            eb.addField("Old Status", old.getMessage(), true);
+                            eb.addField("New Status", guild.getReportManager().getReport(id).getStatus().getMessage(), true);
+                            eb.addField("Responsible moderator", sender.getAsMention(), true);
```
Comment (WalshyDev) [inline src/main/java/stream/flarebot/flarebot/commands/moderation/ReportsCommand.java]: Just chain these, I see no reason to make a new builder and keep doing this. You can easily new line each addField anyway as a chain. Builders are builders for a reason

File: src/main/java/stream/flarebot/flarebot/commands/general/ReportCommand.java [hunk 2]
```diff
@@ -50,6 +52,8 @@ public void onCommand(User sender, GuildWrapper guild, TextChannel channel, Mess
 
             guild.getReportManager().report(report);
 
+            guild.getAutoModConfig().postToModLog(user, sender, new Punishment(ModlogAction.REPORT_CREATE), MessageUtils.getMessage(args, 1));
```
Comment (WalshyDev) [inline src/main/java/stream/flarebot/flarebot/commands/general/ReportCommand.java]: This shows the reportee as the "responsible moderator" may be best to use the EmbedBuilder specific way of doing this.

File: src/main/java/stream/flarebot/flarebot/commands/moderation/ReportsCommand.java (hunk 1 above)
Comment (weeryan17) [reply src/main/java/stream/flarebot/flarebot/commands/moderation/ReportsCommand.java]: I don't see a good way to chain them. if you could send me what your thinking that whould be helpfull

File: src/main/java/stream/flarebot/flarebot/commands/general/ReportCommand.java (hunk 2 above)
Comment (weeryan17) [reply src/main/java/stream/flarebot/flarebot/commands/general/ReportCommand.java]: I changed it. take a look and see what you think

File: src/main/java/stream/flarebot/flarebot/commands/moderation/ReportsCommand.java (hunk 1 above)
Comment (weeryan17) [reply src/main/java/stream/flarebot/flarebot/commands/moderation/ReportsCommand.java]: nvm I'm stupid

File: src/main/java/stream/flarebot/flarebot/commands/moderation/ReportsCommand.java [hunk 3]
```diff
@@ -114,8 +115,17 @@ public void onCommand(User sender, GuildWrapper guild, TextChannel channel, Mess
                         if (guild.getReportManager().getReport(id).getStatus().equals(status)) {
                             MessageUtils.sendInfoMessage("Current status is: **" + status.getMessage() + "**", channel, sender);
                         } else {
+                            ReportStatus old = guild.getReportManager().getReport(id).getStatus();
                             guild.getReportManager().getReport(id).setStatus(status);
                             MessageUtils.sendSuccessMessage(String.format("Changed status of Report with ID: **%d** to **%s**", id, status.getMessage()), channel, sender);
+                            EmbedBuilder eb = new EmbedBuilder()
```
Comment (WalshyDev) [inline src/main/java/stream/flarebot/flarebot/commands/moderation/ReportsCommand.java]: You could just inline, save a variable just being destroyed anyway

File: src/main/java/stream/flarebot/flarebot/mod/Punishment.java [hunk 4]
```diff
@@ -43,7 +43,9 @@ public MessageEmbed getActionEmbed(User user, User responsible, String reason, b
         eb.setTitle(getName());
         eb.setColor(Color.WHITE);
         eb.addField("User", user.getName() + "#" + user.getDiscriminator() + " (" + user.getId() + ")", true);
-        if (responsible != null)
+        if(action == ModlogAction.REPORT_CREATE)
```
Comment (WalshyDev) [inline src/main/java/stream/flarebot/flarebot/mod/Punishment.java]: If we did this for everything the code would be a mass of ifs. Either find a better way to handle this or use the EmbedBuilder option.

Comment (weeryan17) [reply src/main/java/stream/flarebot/flarebot/mod/Punishment.java]: Fair enough

File: src/main/java/stream/flarebot/flarebot/mod/ModlogAction.java [hunk 5]
```diff
@@ -13,7 +13,9 @@
     BAN,
     UNBAN,
 
-    WARN;
+    WARN,
+
+    REPORT_CREATE;
```
Comment (WalshyDev) [inline src/main/java/stream/flarebot/flarebot/mod/ModlogAction.java]: Remove the unused enum and I'll merge

**Pull request merged**